File: guess_number.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void) {
    int myNum;
    int r = rand() % 100 + 1;
    int attempts = 0;

    srand(time(NULL));



    do {
        printf("%s", "Guess a number between 1 and 100: ");
        scanf("%d", &myNum);
        attempts++;

        if (myNum > r) {
            printf("%s\n", "Your number is higher");
        } else if (myNum < r) {
            printf("%s\n", "Your number is lower");
        } else {
            printf("Yay you guessed the number in %d attempts!\n", attempts);
            break;
        }
    } while (myNum != r);

    return 0;
}

File: guess_number.py
import random

random_number = random.randint(1, 100)
attempts = 0

while True:
    user_guess = int(input("Guess a number between 1 and 100: "))
    print(random_number)
    if user_guess > random_number:
        print('Your number is higher than the random number...')
    elif user_guess < random_number:
        print('Your number is lower than the random number...')
    elif user_guess == random_number:
        print(f'Yay you guessed the number with {attempts} attempts :)')
        break
    else:
        print('There was a problem during the program... ')
        break
    attempts += 1